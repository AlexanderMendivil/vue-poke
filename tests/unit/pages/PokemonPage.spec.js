import { shallowMount, mount } from "@vue/test-utils"
import PokemonPageVue from "@/pages/PokemonPage.vue"
import { mockPokemons } from "../mocks/pokemonsMock"
describe('PokemonPage component', () => { 

    let wrapper
    beforeEach( ()=> {

        wrapper = shallowMount( PokemonPageVue )
        jest.clearAllMocks()

    })

    test('should first', () => { 
        expect(wrapper.html()).toMatchSnapshot()
     })

    test('Debe de llamar el mixPokemonArray al montar', () => {
        const mixPokemonArraySpy = jest.spyOn( PokemonPageVue.methods, 'mixPokemonArray' )

        const wrapper = shallowMount(PokemonPageVue)
        expect( mixPokemonArraySpy ).toHaveBeenCalled() 
      })

    test('Debe de hacer match con el snapshot cuando carguen los pokemons', () => {

        const wrapper = mount(PokemonPageVue,{
            data(){
                return{
                    pokemonArr: mockPokemons,
                    pokemon: mockPokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: '',
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
      })

 })