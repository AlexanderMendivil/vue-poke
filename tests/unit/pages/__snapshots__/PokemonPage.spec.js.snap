// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage component Debe de hacer match con el snapshot cuando carguen los pokemons 1`] = `
<div>
  <h1>Â¿Quien es este pokemon?</h1>
  <div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/5.svg" class="hidden-pokemon" alt="pokemon">
    <!--v-if-->
  </div>
  <div class="options-container">
    <ul>
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;

exports[`PokemonPage component should first 1`] = `<h1>Espere por favor...</h1>`;
